modules/search-pipeline-common/src/test/java/org/opensearch/search/pipeline/common/SortResponseProcessorTests.java

```diff
@@ -0,0 +1,230 @@
+
+        assertEquals(response.getHits(), sortResponse.getHits());
+        assertEquals(List.of(9, 6, 5, 4, 3, 2, 1, 1), sortResponse.getHits().getHits()[0].field("digits").getValues());
+        assertEquals(List.of(8, 8, 8, 7, 2, 2, 1, 1), sortResponse.getHits().getHits()[1].field("digits").getValues());
+    }
+
+    public void testSortResponseNullListEntry() {
+        SearchRequest request = createDummyRequest();
+
+        SortResponseProcessor sortResponseProcessor = new SortResponseProcessor(
+            null,
+            null,
+            false,
+            "digits",
+            SortResponseProcessor.SortOrder.ASCENDING,
+            null
+        );
+        assertThrows(
+            IllegalArgumentException.class,
+            () -> sortResponseProcessor.processResponse(request, createTestResponseNullListEntry())
+        );
+    }
+
+    public void testNullField() {
+        SearchRequest request = createDummyRequest();
+
+        SortResponseProcessor sortResponseProcessor = new SortResponseProcessor(
+            null,
+            null,
+            false,
+            "digits",
+            SortResponseProcessor.SortOrder.DESCENDING,
+            null
+        );
+
+        assertThrows(IllegalArgumentException.class, () -> sortResponseProcessor.processResponse(request, createTestResponseNullField()));
+    }
+
+    public void testNotComparableField() {
+        SearchRequest request = createDummyRequest();
+
+        SortResponseProcessor sortResponseProcessor = new SortResponseProcessor(
+            null,
+            null,
+            false,
+            "maps",
+            SortResponseProcessor.SortOrder.ASCENDING,
+            null
+        );
+
+        assertThrows(
+            IllegalArgumentException.class,
+            () -> sortResponseProcessor.processResponse(request, createTestResponseNotComparable())
+        );
+    }
+
+    public void testFactory() {
+        String sortField = RandomDocumentPicks.randomFieldName(random());
+        String targetField = RandomDocumentPicks.randomFieldName(random());
+        Map<String, Object> config = new HashMap<>();
+        config.put("field", sortField);
+        config.put("order", "desc");
+        config.put("target_field", targetField);
+
+        SortResponseProcessor.Factory factory = new SortResponseProcessor.Factory();
+        SortResponseProcessor processor = factory.create(Collections.emptyMap(), null, null, false, config, null);
+        assertEquals("sort", processor.getType());
+        assertEquals(sortField, processor.getSortField());
+        assertEquals(targetField, processor.getTargetField());
+        assertEquals(SortResponseProcessor.SortOrder.DESCENDING, processor.getSortOrder());
+
+        expectThrows(
+            OpenSearchParseException.class,
+            () -> factory.create(Collections.emptyMap(), null, null, false, Collections.emptyMap(), null)
+        );
+    }
+}
```

Comment:
Do we need more tests for different data types sort as well? For eg: SortFloat, SortDoubles etc?

Reply:
any class that implements `Comparable` (which all the `Number` classes do, as well as `String`) will work with the existing code, so the underlying class type doesn't matter.